nekoyume/Assets/_Scripts/UI/Module/GrindModule.cs

```diff
@@ -0,0 +1,324 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Nekoyume.Action;
+using Nekoyume.BlockChain;
+using Nekoyume.Game;
+using Nekoyume.Helper;
+using Nekoyume.L10n;
+using Nekoyume.Model.Item;
+using Nekoyume.Model.Mail;
+using Nekoyume.State;
+using Nekoyume.State.Subjects;
+using Nekoyume.UI.Model;
+using Nekoyume.UI.Scroller;
+using TMPro;
+using UnityEngine;
+
+namespace Nekoyume.UI.Module
+{
+    using UniRx;
+    public class GrindModule : MonoBehaviour
+    {
+        [SerializeField]
+        private Inventory grindInventory;
+
+        [SerializeField]
+        private ConditionalCostButton grindButton;
+
+        [SerializeField]
+        private StakingBonus stakingBonus;
+
+        [SerializeField]
+        private GameObject stakingBonusDisabledObject;
+
+        [SerializeField]
+        private List<GrindingItemSlot> itemSlots;
+
+        // TODO: It is used when NCG can be obtained through grinding later.
+        [SerializeField]
+        private GameObject ncgRewardObject;
+
+        [SerializeField]
+        private TMP_Text ncgRewardText;
+
+        [SerializeField]
+        private TMP_Text crystalRewardText;
+
+        private bool _isInitialized;
+
+        private string _cachedGrindingRewardCrystal;
+
+        private readonly ReactiveCollection<InventoryItem> _selectedItemsForGrind =
+            new ReactiveCollection<InventoryItem>();
+
+        private const int LimitGrindingCount = 10;
+
+        private bool CanGrind => _selectedItemsForGrind.Any() &&
+                                 _selectedItemsForGrind.All(item => !item.Equipped.Value);
+
+        private void Initialize()
```

Comment:
`Subscribe` streams are still remaining even if this module is not using any more. I highly suggest you to `Subscribe` on every `Show` and dispose it on `OnDisable`

Reply:
I agree. I will reflect it.